### PR TITLE
Add Semantic Versioning Release workflow [Rebase & FF]

### LINKED ISSUE #50: Add process to ensure semantic versioning and release consistency
We can use information already available (labels) to improve our ability to track semantic versioning compliance with more frequent and better documented releases.

This feature should include the ability have automated releases where semantic versioning is rolled according to PR impact data and the next release info is always available to see what's happened since the last release. This makes it easy to determine when to release (on minor and major version revs) and do so with minimal effort.

In addition, the release info should be categorized so it's easier to understand what kind of changes are in a release.

### PR DESCRIPTION
**Pull request by makubacki**
Closes #50

Adds a new workflow and accompanying configuration files that allow
releases to automatically be drafted in repos as pull requests are
completed.

Semantic versioning is automatically determined based on the standard
set of Project Mu labels associated with the pull request.

The types of changes (e.g. bug fix, new feature, etc.) are
automatically categorized in the pull request based on the labels as
well.

This makes tracking the new Semantic Version based on the type of
changes automatic and a release is always ready. In between releases,
the draft is a nice way to see a high-level delta since the last
release.

- `.github/ReleaseDrafter.yml` - Single reusable workflow to maintain
  a release draft as pull requests complete.
- `.github/ReleaseDraft.yml` - Configuration file for the
  ReleaseDrafter workflow. Defines label meanings for releases and
  how releases should be drafted.
- `.sync/workflows/leaf/release-draft.yml` - A leaf workflow that can
  be synced to repos to trigger the reusable workflow in Mu DevOps.

Project Mu repos can easily opt into consistent release drafting by
including this leaf workflow via file sync.

---

I decided to allow dependabot changes into release and just not include
file sync changes since those are dev ops related and already tracked in
the mu_devops releases where they're more relevant.

### CODE REVIEW

**Comment by makubacki** [comment]
Fyi: Holding up other PRs so they can be tracked in versioning after this PR is merged.

File: .github/workflows/ReleaseDrafter.yml
Comment (apop5) [inline .github/workflows/ReleaseDrafter.yml:2]: the first 'will' should be removed.

File: .sync/workflows/leaf/release-draft.yml
Comment (apop5) [inline .sync/workflows/leaf/release-draft.yml:2]: first "will" can be dropped.

File: .github/workflows/ReleaseDrafter.yml
Comment (kenlautner) [inline .github/workflows/ReleaseDrafter.yml:2]:
```suggestion
# what will a current release will look like at any time.
what a current release will look like at any time.
```

**Review by apop5 — APPROVED** [review]

**Review by kenlautner — APPROVED** [review]

File: ReadMe.rst
Comment (Flickdm) [inline ReadMe.rst:137]:
```suggestion
In order to ensure semantic versioning is followed based on well-defined labels used in Project Mu pull requests, the
```

**Review by kuqin12 — APPROVED** [review]

File: .sync/Files.yml
Comment (spbrogan) [inline .sync/Files.yml:358]: does it make sense to push thin into any repo we have a "main"?  

Feature repos, mu_tiano_platforms, etc?

Comment (makubacki) [reply .sync/Files.yml:358]: Yes. Honestly, I wanted to confirm everything in mu_devops for a release or two before doing so. It is tested on my fork but thought I'd double check here to reduce possible sync thrash in case any small tweaks need to be made before syncing elsewhere.

File: ReadMe.rst
Comment (makubacki) [reply ReadMe.rst]: Update in PR commit https://github.com/microsoft/mu_devops/pull/49/commits/b445c98d20c87a5e36f27d8743aa98b44ca2795d

File: .sync/workflows/leaf/release-draft.yml
Comment (makubacki) [reply .sync/workflows/leaf/release-draft.yml]: Updated in PR commit https://github.com/microsoft/mu_devops/pull/49/commits/b445c98d20c87a5e36f27d8743aa98b44ca2795d

File: .github/workflows/ReleaseDrafter.yml
Comment (makubacki) [reply .github/workflows/ReleaseDrafter.yml]: Updated in PR commit https://github.com/microsoft/mu_devops/pull/49/commits/b445c98d20c87a5e36f27d8743aa98b44ca2795d

File: ReadMe.rst
Comment (makubacki) [reply ReadMe.rst]: Updated in PR commit https://github.com/microsoft/mu_devops/pull/49/commits/b445c98d20c87a5e36f27d8743aa98b44ca2795d

File: .github/workflows/ReleaseDrafter.yml
Comment (makubacki) [reply .github/workflows/ReleaseDrafter.yml]: Updated in PR commit https://github.com/microsoft/mu_devops/pull/49/commits/b445c98d20c87a5e36f27d8743aa98b44ca2795d

**Pull request merged**
